wpep0008.md

```diff
@@ -0,0 +1,50 @@
+| Key | Value |
+| ---: | :--- |
+| **Title** | Wazo integration tests framework |
+| **Number** | 0008 |
+| **Date** | 2023-08-28 |
+| **Authors** | Jesse Sopel <jesse.sopel@wazo.io>, Sebastien Duthil <sduthil@wazo.io> |
+| **Status** | draft |
+
+## Abstract
+
+Wazo integration tests: Explicit the direction of evolution of integration tests
+
+## Motivation
+
+The integration test framework was implemented in Wazo before the usage of
+pytest. Pytest offers a very Pythonic way to organize tests and fixtures, and we
+want to use this system more and more, since it offers more flexibility,
+readability, performance and reusability.
+
+## State of the art in Wazo Platform
+
+Integration tests are organised around `AssetLaunchingTestCase`, inheriting from
+the `unittest` testing framework. This organization forces tests to be organized
+around a test class, which may not be as flexible as we want.
```

Comment:
A minimal example of such a typical test case would be nice(with some setUp/teardown) to make things explicit.

Reply:
I added the example provided by Jesse below.